### DISCUSSION TITLE
Set relative height/width in heatmap

### DISCUSSION THREAD

**Discussion by hcraT** [vega/vega-lite#7314] [Q&A] [answered]
Hi,
I would like to be able to set the width of the rect marks of an heatmap in a way that is independent from the chart size.

From the documentation I figured out how to set the absolute size:
```json
{
  "$schema": "https://vega.github.io/schema/vega-lite/v5.json",
  "data": {"url": "data/cars.json"},
  "mark": {
    "type": "rect",
    "width": 10
  },
  "encoding": {
    "y": {"field": "Origin", "type": "nominal"},
    "x": {"field": "Cylinders", "type": "ordinal"},
    "color": {"aggregate": "mean", "field": "Horsepower"}
  },
  "config": {
    "axis": {"grid": true, "tickBand": "extent"}
  }
}
```
It is not possible, however, to specify a value like "50%" for the width field. With 50% I mean that each cell of the heatmap is centered in the x coordinate and fills the 50% of the available width (instead of the default 100%).

I know that one could set both the `x` and `x2` fields and I'm using this approach. So I compute the coordinates of the rects manually but this means that I need to override the axes as well to properly label my heatmap. From the documentation I wasn't able to figure out if there are options to obtain what I need.

Do you have suggestions?

Thank you very much.

Best!

**Comment by domoritz** [comment, marked as answer]
I think you can achieve this with the [Band scale properties](https://vega.github.io/vega-lite/docs/scale.html#band). Something like [this](https://vega.github.io/editor/#/url/vega-lite/N4IgJAzgxgFgpgWwIYgFwhgF0wBwqgegIDc4BzJAOjIEtMYBXAI0poHsDp5kTykBaADZ04JAKyUAVhDYA7EABoQAEySYUqUAwBOgtCrVICUJNohSZ8gL5Lk2gNZpQmAJ444+7XCiYQNkHCyUGzKNLJkTiAukQBmNHCCyvoA8to0tPJKru76smwIYUh6-gAesfGJ+gDCLsKyynBmiiDZHuhs2qGyRc3QRW2gOEjKXRGoAAyUYkpF6fKoAIxW-sGCHZFIZGReFJhtIAhwSJkgcQlJ6AASHRBwOGwA7o1+K3JxY6BIJTQQkds0F0w2gYcCyNCg9gAQscLgESntZL5llYgA)


```json
...
"scale": {"padding": 0.5, "align": 1}
...
```

**Reply by hcraT** [reply]
Hi,
thanks for your reply.
This indeed works.

When I try to apply this to my use case I get a visualization issue (the x axis moves to the top of the plot). See [here](https://vega.github.io/editor/#/url/vega-lite/N4IgJAzgxgFgpgWwIYgFwhgF0wBwqgegIDc4BzJAOjIEtMYBXAI0poHsDp5kTykBaADZ04JAKyUAVhDYA7EABoQAEySYUqUAwBOgtCrVICUJNohSZ8gL5LBSAJ5xtaANqhk2gNb7tcKJkUQOFkoNmUaWTI0UHtokAAzGjhBZX0AeW0aWnklTHscOH1ZNgQIpD0laHLCzRAcJGVwyLQABisbEAAPOMTk1PQAYXthWWUnCEC8gv02bSbykA7QwVm4pDIyXwpMGpBSnISklP0ACVmIOBw2AHcnRfaFd1NvWqnd339AiExtNk93uCpDrBUJNKK1WK1XrHdAZLIRSb5XbFfYLSomQQ1UD1RoRcEtShiJTlLLyVAEgBMYg6SE6NAmrxoUE8DPi5QutiQTGSrPZcFydExaFkDEEggeXR6R36ICGIzGZkR03Qs3meiWbBWzlq602fB2+mQ3SU0JlZzMlxud3aVgAukpfDJBKQ4lUhRD9BExgVRsEAkplqt0F7LsExrIAjagA) for the example (the plot is nonsensical just to showcase my need).

Is this a bug? Should I open an issue?

I played around a little bit and I think that the problem is due to the `resolve` config because it happens even if [I don't specify any scale](https://vega.github.io/editor/#/url/vega-lite/N4IgJAzgxgFgpgWwIYgFwhgF0wBwqgegIDc4BzJAOjIEtMYBXAI0poHsDp5kTykBaADZ04JAKyUAVhDYA7EABoQAEySYUqUAwBOgtCrVICUJNohSZ8gL5LBSAJ5xtaANqhk2gNb7tcKJkUQOFkoNmUaWTI0UHtokAAzGjhBZX0AeW0aWnklTHscOH1ZNgQIpD0bEAAPOMTk1PQAYXthWWUnCEC8gv02bXDZcpBK0ME+uKQyMl8KTEL0UpyEpJT9AAk+iDgcNgB3J2GrG3dTb3Rff0Dg0IGozRBY+7rV9AysiK78+ZBixaHKmpPFYNEDNVrtMyfHroPoDf5KUbje6TaZ8Ob6RaBZ4gjZmbZ7A5HKwAXSUvhkglIcWg5XmMX0EXaBTawQCRKAA). Just setting either `resolve.scale.x = "independent"` or `resolve.scale.y = "independent"` messes things up. However if I set any `height` for the plot when set the `y` scale to independent or any `width` when instead acting on the `x` scale the problem goes away (see [here](https://vega.github.io/editor/#/url/vega-lite/N4IgJAzgxgFgpgWwIYgFwhgF0wBwqgegIDc4BzJAOjIEtMYBXAI0poHsDp5kTykBaADZ04JAKyUAVhDYA7EABoMcGmSxoAjAAYtSgCZJMKVKAYAnQWhAGjBKEjMQpM+QF8lgpAE84ZtAG1QZDMAayszOChMRRA4WSg2PRpZMjRQLzSQADMaOEE9KwB5M1VkmMwvHDgrWTYEZKRLdxAAD0ycvIL0AGEvYVk9Xwhyyur0NjMk2UaQZoTBCcykMjIIikwxkHr5JQ78qwAJCYg4HDYAd19Z13cghzD0CKiYuISp1JMQDM+9rpBi0o7EAVKo1OoNJpKNo-XL7Hp9ZKDRwjUHjSYQ2ZKeaLT7LVZ8DZWbYxX6HY6nC5XG6uAC6SgiMkEpEy0EaY3SVkRpzig1k0WpQA)).

Thanks!
Best
